src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -32,10 +33,27 @@ public class DefaultModelMetadata : ModelMetadata
         /// <param name="provider">The <see cref="IModelMetadataProvider"/>.</param>
         /// <param name="detailsProvider">The <see cref="ICompositeMetadataDetailsProvider"/>.</param>
         /// <param name="details">The <see cref="DefaultMetadataDetails"/>.</param>
+        /// <remarks>Use other constructor overload. This is provided for back compatibility and a few tests.</remarks>
```

Comment:
I don't really feel like this is necessary or accurate. The API shape makes it clear what the tradeoff is here.